### PR TITLE
replace multi yaml file parsing via yaml separator

### PR DESCRIPTION
**Pull request by HiteshRepo**
*Issue #, if available:*
Yaml separator '---' was used to split and read a multi yaml manifests.
The yaml manifests might contain certificate data which might lead to erroneous results.

*Description of changes:*
Replacing the parsing logic using yaml util library ("k8s.io/apimachinery/pkg/util/yaml")

### CODE REVIEW

**Comment by HiteshRepo** [comment]
@chrisdoherty4 could you please take a look at this?

File: pkg/api/v1alpha1/tinkerbellmachineconfig.go [hunk 1]
```diff
@@ -2,10 +2,10 @@ package v1alpha1
 
 import (
 	"fmt"
-	"os"
-	"strings"
 
+	yamlutilpkg "github.com/aws/eks-anywhere/pkg/yamlutil"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
```
Comment (chrisdoherty4) [inline pkg/api/v1alpha1/tinkerbellmachineconfig.go:8]:
```suggestion
	apimachineryyaml "k8s.io/apimachinery/pkg/util/yaml"
```

File: pkg/yamlutil/parser.go [hunk 2]
```diff
@@ -174,3 +176,28 @@ func (p *Parser) unmarshal(reader io.Reader) (*parsed, error) {
 func (p *Parser) buildConfigFromParsed(parsed *parsed, b Builder) error {
 	return b.BuildFromParsed(parsed.objects)
 }
+
+func ParseMultiYamlFile(fileName string) ([][]byte, error) {
```
Comment (chrisdoherty4) [inline pkg/yamlutil/parser.go:180]: Given this is a generic YAML parsing function it ought to live in `/pkg/utils/yaml`. This package, contrary to its name, isn't a set of generic YAML utils - apologies for the confusion.

Comment (chrisdoherty4) [inline pkg/yamlutil/parser.go:180]: Can we accept an `io.Reader`? It makes the function compatible with non-file sources and simplifies testing as files aren't necessary.

The function can also be simplified to `SplitDocuments` particularly considering we aren't really parsing the file.

```go
func SplitDocuments(r io.Reader) ([][]byte, error) { ... }
```

```go
func main() {
	reader := ...
	yaml.SplitDocuments(reader)
}
```

File: pkg/api/v1alpha1/tinkerbellmachineconfig.go [hunk 3]
```diff
@@ -2,10 +2,10 @@ package v1alpha1
 
 import (
 	"fmt"
-	"os"
-	"strings"
 
+	yamlutilpkg "github.com/aws/eks-anywhere/pkg/yamlutil"
```
Comment (chrisdoherty4) [inline pkg/api/v1alpha1/tinkerbellmachineconfig.go:6]:
```suggestion
	"github.com/aws/eks-anywhere/pkg/yamlutil"
```

**Review by chrisdoherty4 — CHANGES_REQUESTED** [review]
Thank you for the submission @HiteshRepo. Apologies for the delay, again. These are great improvements.

**Comment by chrisdoherty4** [comment]
Hi @HiteshRepo. Thank you for the changes, functionally everything looks great. The CI is complaining about a lack of function code comments and a compilation error (I think). Once that's fixed up this is good to go!

**Pull request merged**